runtime/connectors/gcs/gcs.go

```diff
@@ -86,10 +88,19 @@ func (c connector) ConsumeAsIterator(ctx context.Context, env *connectors.Env, s
 		return nil, fmt.Errorf("invalid gcs path %q, should start with gs://", conf.Path)
 	}
 
-	bucket := fmt.Sprintf("gs://%s", url.Host)
-	bucketObj, err := blob.OpenBucket(ctx, bucket)
+	creds, err := resolvedCredentials(ctx, env)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := gcp.NewHTTPClient(gcp.DefaultTransport(), gcp.CredentialsTokenSource(creds))
```

Comment:
Is `gcp.CredentialsTokenSource` guaranteed to work for all kinds of credentials? Looking at the implementation, it could appear it only works for token creds

Reply:
Yes the credentials returned takes care of all kind of credentials : https://github.com/golang/oauth2/blob/c82d0e16dcd7c2b3907866fc2124725806a3ed67/google/default.go#L192

Reply:
Perfect, thanks!